RestAPI/Controllers/EmployeeController.cs

```diff
@@ -0,0 +1,160 @@
+            }
+            existingEmployee.FirstName = request.FirstName;
+            existingEmployee.LastName = request.LastName;
+            existingEmployee.WorkingNumber = request.WorkingNumber;
+            existingEmployee.Email = request.Email;
+            //existingEmployee.Shifts = EmployeeGrpcRepository.EntityShiftDtosToEntityShiftsList(request.Shifts);
+            //existingEmployee.Id = request.Id;
+            existingEmployee.Password = AuthController.Hash(request.Password);
+
+            Employee updated = await employeeRepo.UpdateAsync(existingEmployee);
+            
+            PublicEmployeeDTO dto = new()
+            {
+                FirstName = updated.FirstName,
+                LastName = updated.LastName,
+                WorkingNumber = updated.WorkingNumber,
+                Id = updated.Id
+                
+            };
+            return Accepted($"/Employee/{dto.Id}", $"{dto.FirstName} {dto.LastName} id=[{dto.Id}] was updated!");
+        }
+        catch (ArgumentException e)
+        {
+            return BadRequest(e.Message);
+        }
+    }
+    
+    
+    
+    [HttpGet("/Employee/{id:long}")]
+    public async Task<ActionResult<ShiftEmpoyeeDTO>> GetSingle([FromRoute] long id)
+    {
+       // Console.WriteLine(id.GetType());
+        try
+        {
+            Employee gotEmployee = await employeeRepo.GetSingleAsync(id);
+            Console.WriteLine("ecntrl-get single eShifts: " + gotEmployee.PrintShifts());
+            
+            ShiftEmpoyeeDTO dto = new()
+            {
+                WorkingNumber = gotEmployee.WorkingNumber,
+                FirstName = gotEmployee.FirstName,
+                LastName =  gotEmployee.LastName,
+                Id = gotEmployee.Id,
+                Shifts = EmployeeGrpcRepository.EntityShiftsToEntityShiftDTOs(gotEmployee.Shifts),
+            };
+            return Accepted($"/Employee/{gotEmployee.Id}", dto);
+        }
+        catch (ArgumentException e)
+        {
+            return BadRequest(e.Message); 
+        }
+        
+    }
+    
+    [HttpGet("/Employees/")]
+    public async Task<ActionResult<List<PublicEmployeeDTO>>> GetMany()
+    {
+        try
+        {
+            Console.WriteLine("1");
+            IQueryable<Employee> employees = employeeRepo.GetManyAsync();
+            Console.WriteLine("2");
+
+            List<PublicEmployeeDTO> dtos = employees.Select(employee => new PublicEmployeeDTO
+            {
+                WorkingNumber = employee.WorkingNumber,
+                FirstName = employee.FirstName,
+                LastName = employee.LastName,
+                Id = employee.Id
+            }).ToList();
+            Console.WriteLine("3");
+
+            return Ok(dtos);
+        }
+        catch (Exception e)
+        {
+            return Problem(e.Message); 
+        }
+    }
+    
+    [HttpDelete("{id:long}")]
+    public async Task<ActionResult> Delete([FromRoute] long id)
```

Comment:
Does this one delete an employee? a little unsure because no Employee in route

Reply:
we are deleting based on the id which is unique